### PR TITLE
add validation on required_course_count in course group model

### PR DESCRIPTION
**Pull request by yaelmualem**
added validation on required_course_count  - checked it is a non negative value

### CODE REVIEW

**Comment by ScDor** [comment]
How does the FE handle such exceptions?

**Comment by yaelmualem** [comment]
it translated to 'BadRequest':
image
image

**Comment by ScDor** [comment]
@rbenjos @noabarlia Seems like it requires supporting such exceptions in the FE before merging. Wdyt?

**Comment by rbenjos** [comment]
i dont think it has anything to do with us. when a request is "bad", we just don't present any results.

**Comment by ScDor** [comment]
Ok cool. @yaelmualem please solve the remaining conflicts and merge away

**Comment by yaelmualem** [comment]
@ScDor I cant approve my own PR, can you arrive it?

File: Closure_Project/rest_api/models.py [hunk 1]
```diff
@@ -149,7 +161,7 @@ class CourseGroup(models.Model):
     course_type = models.CharField(max_length=20, choices=CourseType.choices)
     year_in_studies = models.IntegerField(choices=Year.choices)
     index_in_track_year = models.IntegerField()
-    required_course_count = models.IntegerField(null=True)
+    required_course_count = models.IntegerField(null=True, validators=[_validate_non_negative_number])
     required_points = models.IntegerField(null=True)
```
Comment (ScDor) [inline Closure_Project/rest_api/models.py:165]: Could we use the save validation for other fields that are integers, such as `points required`? Note some are nullable

Comment (yaelmualem) [reply Closure_Project/rest_api/models.py]: added to the other fields that aren't the year. should I check there larger than 2021?

Comment (ScDor) [reply Closure_Project/rest_api/models.py]: Nah, let's not make our future lives (in two months, when 2022 is out) too miserable for no real reason

Comment (yaelmualem) [reply Closure_Project/rest_api/models.py]: so approve?

Comment (ScDor) [reply Closure_Project/rest_api/models.py]: Idk as you assigned @ilanerukh to do it, but I'll take the liberty to assign myself and approve it (:

**Review by ScDor — APPROVED** [review]

**Pull request merged**